fgpyo/sam/__init__.py

```diff
@@ -849,16 +849,24 @@ def from_read(cls, read: pysam.AlignedSegment) -> List["SupplementaryAlignment"]
 def sum_of_base_qualities(rec: AlignedSegment, min_quality_score: int = 15) -> int:
     """Calculate the sum of base qualities score for an alignment record.
 
-    This function is useful for calculating the "mate score" as implemented in samtools fixmate.
+    This function is useful for calculating the "mate score" as implemented in `samtools fixmate`.
+    Consistently with `samtools fixmate`, this function returns 0 if the record has no base
+    qualities.
 
     Args:
         rec: The alignment record to calculate the sum of base qualities from.
         min_quality_score: The minimum base quality score to use for summation.
 
+    Returns:
+        The sum of base qualities on the input record. 0 if the record has no base qualities.
+
     See:
         [`calc_sum_of_base_qualities()`](https://github.com/samtools/samtools/blob/4f3a7397a1f841020074c0048c503a01a52d5fa2/bam_mate.c#L227-L238)
         [`MD_MIN_QUALITY`](https://github.com/samtools/samtools/blob/4f3a7397a1f841020074c0048c503a01a52d5fa2/bam_mate.c#L42)
     """
+    if rec.query_qualities is None:
+        return 0
```

Comment:
For safety-sake, such as if someone manually created a SAM record with the `*`, I'd probably still want an extra guard here:

```suggestion
    if rec.query_qualities is None or rec.query_qualities == NO_QUERY_QUALS:
        return 0
```

Reply:
Additionally, after assigning to `record.query_sequence`, the `query_qualities` is automatically set to `None`

https://github.com/pysam-developers/pysam/blob/0eae5be21ac3ab3ac7aa770a3931e2977e37b909/pysam/libcalignedsegment.pyx#L1461-L1463

```py
In [10]: record.query_qualities

In [11]: record = builder.add_single()

In [12]: record.query_sequence = "*"

In [13]: record.query_qualities is None
Out[13]: True
```

Reply:
It's deprecated, but you can use `single.qual = None` or `single.qual = ""` or `single.qual = "*"`.

Reply:
https://github.com/pysam-developers/pysam/pull/1324

Reply:
Ah, I see. 

Then I think the correct thing to do is to check if the record's `query_qualities` is equal to `qualitystring_to_array(NO_QUERY_QUALS)`, not the string itself.